### PR TITLE
optimization debian package manager tweaks

### PR DESCRIPTION
**Pull request by rajpratik71**
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

### CODE REVIEW

**Comment by rajpratik71** [comment]
@fud @akyrtzi @lgaches  requesting review

**Comment by akyrtzi** [comment]
\cc @shahmishal

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@swift-ci test

File: 5.0/ubuntu/16.04/slim/Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ LABEL maintainer="Swift Infrastructure <swift-infrastructure@swift.org>"
 LABEL Description="Docker Container for the Swift programming language"
 
 RUN export DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true && apt-get -q update && \
-    apt-get -q install -y \
+    apt-get --no-install-recommends install -y apt-utils ca-certificates \
```
Comment (ianpartridge) [inline 5.0/ubuntu/16.04/slim/Dockerfile]: Please put the `apt-utils` and `ca-certificates` on their own line, to match the rest of the packages.

**Review by ianpartridge — COMMENTED** [review]
In principle, great idea!

How many unnecessary packages does this remove, and what effect does it have on final image size?

**Comment by tomerd** [comment]
hi @Rajpratik71, would you like to work with us to bring this over the finish line?

**Comment by rajpratik71** [comment]
> hi @Rajpratik71, would you like to work with us to bring this over the finish line?

Hi , rebased to upstream and also done changes as requested by @ianpartridge

**Review by ianpartridge — APPROVED** [review]

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by tomerd** [comment]
thanks so much @Rajpratik71, can you remind me why we needed to add the `apt-utils` and `ca-certificates` packages as part of this PR?

additionally, we recently added support for Ubuntu 20.04 (5.2 and nightlies) which I think can use the same optimization?

**Comment by rajpratik71** [comment]
> thanks so much @Rajpratik71, can you remind me why we needed to add the `apt-utils` and `ca-certificates` packages as part of this PR?
> 
> additionally, we recently added support for Ubuntu 20.04 (5.2 and nightlies) which I think can use the same optimization?

added packages apt-utils ca-certificates

Because build is

1.  Slow because "apt-utils" not installed

2. to avoid build to exits with error without having certificate in wget , curl or even git clone

**Comment by tomerd** [comment]
hi @Rajpratik71

1. for my education, how did it work before this change? were they part of recommend packages?

2. could you please apply the same logic/patch to the recently added 20.04 images

thanks a ton for your contribution!

**Comment by rajpratik71** [comment]
> hi @Rajpratik71
> 
> 1. for my education, how did it work before this change? were they part of recommend packages?
> 2. could you please apply the same logic/patch to the recently added 20.04 images
> 
> thanks a ton for your contribution!

Hi @tomerd ,

These packages are not there before . e.g If any debian based system doesn't have "apt-utils" installed , then in that case any package installation takes time while configuring and gives suggestion to install "apt-utils" . that is why i put that here  .

And I put "ca-certificates" to avoid build to exits with error without having certificate in wget , curl or even git clone.

And yes , for 20.04 images will do that

**Comment by tomerd** [comment]
@swift-ci Please test

**Comment by shahmishal** [comment]
The Swift project moved the default branch to `main` and deleted `master` branch, so GitHub automatically closed the PR. Please re-create the pull request with `main` branch.

More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

**Comment by rajpratik71** [comment]
> The Swift project moved the default branch to `main` and deleted `master` branch, so GitHub automatically closed the PR. Please re-create the pull request with `main` branch.
> 
> More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

created fresh pull to main @ #203

**Review by albertoblue87-netizen — COMMENTED** [review]
Done

**Pull request closed without merging**